packages/wrangler/src/__tests__/pages.test.ts

```diff
@@ -2543,58 +2543,55 @@ and that at least one include rule is provided.
 			);
 
 			await runWrangler("pages project upload .");
-
 			expect(requests.length).toBe(3);
 
-			const resolvedRequests = await Promise.all(
-				requests.map(async (req) => await req.json<UploadPayloadFile[]>())
-			);
-
-			const sortedRequests = resolvedRequests.sort((a, b) => {
-				const aKey = a[0].key;
-				const bKey = b[0].key;
+			const resolvedRequests = (
+				await Promise.all(
+					requests.map(async (req) => await req.json<UploadPayloadFile[]>())
+				)
+			).flat();
 
-				return aKey.localeCompare(bKey);
-			});
+			const requestMap = resolvedRequests.reduce<{
+				[key: string]: UploadPayloadFile;
+			}>(
+				(requestMap, req) => Object.assign(requestMap, { [req.key]: req }),
+				{}
+			);
 
 			for (const req of requests) {
 				expect(req.headers.get("Authorization")).toBe(
 					"Bearer <<funfetti-auth-jwt>>"
 				);
 			}
 
-			expect(sortedRequests[0]).toMatchObject([
-				{
-					base64: true,
-					key: "09a79777abda8ccc8bdd51dd3ff8e9e9",
-					metadata: {
-						contentType: "application/javascript",
-					},
-					value: "ZnVuYw==",
+			expect(Object.keys(requestMap).length).toBe(3);
+
+			expect(requestMap["95dedb64e6d4940fc2e0f11f711cc2f4"]).toMatchObject({
```

Comment:
@GregBrimble I found it more reliable to not sort the requests and lookup by index. But rather key the requests by the `key` property and lookup by said `key` property